lib/core/test/unit/Cardano/Wallet/Primitive/Types/UTxOIndexSpec.hs

```diff
@@ -505,7 +508,7 @@ prop_selectRandom_one_withAdaOnly u = checkCoverage $ monadicIO $ do
 -- This should only succeed if there is at least one element with a non-zero
 -- quantity of the asset.
 --
-prop_selectRandom_one_withAsset :: UTxOIndex -> AssetId -> Property
+prop_selectRandom_one_withAsset :: UTxOIndex InputId -> AssetId -> Property
```

Comment:
> All these properties would work with any Ord i => UTxOIndex i right?

Yes, precisely!

I haven't converted the test suites yet, but I'm thinking of making a simple `TestUTxOKey` type with generators and shrinkers for it.

I haven't done this yet, because I predict some effort (though not too much) with be required to maintain the coverage conditions.